usaspending_api/search/v2/views/spending_by_category_views/spending_by_recipient.py

```diff
@@ -64,18 +63,35 @@ def _get_recipient_id(row: list) -> str:
         )
 
     def build_elasticsearch_result(self, response: dict) -> List[dict]:
+        # Get the codes
+        recipient_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
+        recipient_hashes = [bucket.get("key").split("/")[0] for bucket in recipient_info_buckets if bucket.get("key")]
+
+        # Get the current recipient info
+        current_recipient_info = {}
+        recipient_info_query = RecipientLookup.objects.filter(recipient_hash__in=recipient_hashes).values(
+            "duns", "legal_business_name", "recipient_hash"
+        )
+        for recipient_info in recipient_info_query.all():
+            current_recipient_info[str(recipient_info["recipient_hash"])] = recipient_info
 
+        # Build out the results
         results = []
-        location_info_buckets = response.get("group_by_agg_key", {}).get("buckets", [])
-        for bucket in location_info_buckets:
-            recipient_info = json.loads(bucket.get("key"))
+        for bucket in recipient_info_buckets:
+            result_hash, result_level = tuple(bucket.get("key").split("/")) if bucket.get("key") else (None, None)
```

Comment:
Just double checking, but would this accurately catch the condition where the key has a hash but not a recipient_level? I believe this situation can happen with some of the "special" recipients since the levels come from a join that excludes them and the hash does not.

Reply:
It's a good call to double-check as it can get funky. I just reviewed the `test_spending_by_recipients.py` which includes scenarios of `MULTIPLE RECIPIENTS` without levels in the transaction search data. The recipient agg key stores them as `[hash]/` in the index and then this would split them into the variables (quick local run of that line):
```
>>> bucket = {'key': 'example_hash/'}
>>> result_hash, result_level = tuple(bucket.get("key").split("/")) if bucket.get("key") else (None, None)
>>> result_hash
'example_hash'
>>> result_level
''
```
And then the reponse has something like:
```
{"amount": 5000.0, "code": "096354360", "name": "MULTIPLE RECIPIENTS", "recipient_id": None}
or
{
     "amount": 5000000.0,
     "code": "Recipient not provided",
    "name": "MULTIPLE RECIPIENTS",
    "recipient_id": None,
},
```
I think this should cover most of those weird cases.